linetools/spectra/io.py

```diff
@@ -252,7 +256,12 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, flux_tags=None,
             uwave = u.Quantity(wave, unit=u.AA)
         else:
             uwave = u.Quantity(wave)
+        # Sort
+        srt = np.argsort(uwave)
+        uwave = uwave[srt]
+        fx = fx[srt]
         if sig is not None:
+            sig = sig[srt]
             xspec1d = XSpectrum1D.from_array(uwave, u.Quantity(fx),
```

Comment:
It might be better to overload `from_array` so that it wraps `Spectrum1D.from_array`, but adds a keyword option `sort` which is false by default. What do you think?